autopilot/externalscoreattach.go

```diff
@@ -0,0 +1,124 @@
+package autopilot
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btcutil"
+)
+
+// ScoreSettable is an interface that indicates that the scores returned by the
+// heuristic can be set by an external caller. The ExternalScoreAttachment
+// imlements this interface, and so should any heuristic that is using the
+// ExternalScoreAttachment as a sub-heuristic, to allow access to setting the
+// internal scores.
+type ScoreSettable interface {
+	// SetNodeScores is used to set the internal map from NodeIDs to
+	// scores. The passed scores must be in the range [0, 1.0].
+	SetNodeScores(map[NodeID]float64) error
+}
+
+// ExternalScoreAttachment is an implementation of the AttachmentHeuristic
+// interface that allows an external source to provide it with node scores.
+type ExternalScoreAttachment struct {
+	nodeScores map[NodeID]float64
+
+	sync.Mutex
+}
+
+// NewExternalScoreAttachment creates a new instance of an
+// ExternalScoreAttachment.
+func NewExternalScoreAttachment() *ExternalScoreAttachment {
+	return &ExternalScoreAttachment{}
+}
+
+// A compile time assertion to ensure ExternalScoreAttachment meets the
+// AttachmentHeuristic and ScoreSettable interfaces.
+var _ AttachmentHeuristic = (*ExternalScoreAttachment)(nil)
+var _ ScoreSettable = (*ExternalScoreAttachment)(nil)
+
+// Name returns the name of this heuristic.
+//
+// NOTE: This is a part of the AttachmentHeuristic interface.
+func (s *ExternalScoreAttachment) Name() string {
+	return "externalscore"
```

Comment:
Not a blocker, but would be nice if you could name external scores for debugging purposes, e.g. `”externalscore-<name>”`

Reply:
This can also be used to target a specific external score if multiple are used in the weighted combination